### PR TITLE
Grant LEVELBUILDERs abilities.

### PR DESCRIPTION
**Pull request by ashercodeorg**
The intent of this PR is to allow separation of access to levelbuilder (as a levelbuilder) from the admin permission.

### CODE REVIEW

File: dashboard/app/models/ability.rb [hunk 1]
```diff
@@ -166,6 +166,17 @@ def initialize(user)
...
+    if user.permission?(UserPermission::LEVELBUILDER)
+      can :manage, [
+        FrequentUnsuccessfulLevelSource,
+        Level,
+        LevelSourceHint,
+        Script,
+        ScriptLevel,
+        Stage,
+      ]
+    end
```
Comment (Hamms) [inline dashboard/app/models/ability.rb:183]: do we need to add the same `cannot [:update, :destroy], Level do |level| !level.custom? end` addendum as we do for admins?

File: dashboard/app/models/ability.rb [hunk 2]
```diff
@@ -166,6 +166,17 @@ def initialize(user)
       end
     end
 
+    if user.permission?(UserPermission::LEVELBUILDER)
+      can :manage, [
+        FrequentUnsuccessfulLevelSource,
+        Level,
+        LevelSourceHint,
```
Comment (Hamms) [inline dashboard/app/models/ability.rb]: why is this in here?

File: dashboard/app/models/ability.rb (hunk 1 above)
Comment (ashercodeorg) [inline dashboard/app/models/ability.rb:183]: Done.

File: dashboard/app/models/ability.rb (hunk 2 above)
Comment (ashercodeorg) [inline dashboard/app/models/ability.rb]: Removed. Also FrequesntUnsuccessfulLevelSource. Though I don't see why we shouldn't give LBs those abilities.

**Comment by ashercodeorg** [comment]
Note that the failing tests seem legitimate. Not only can I reproduce locally, but reverting the change makes the errors disappear. But I have no understanding how my change touches the failing tests. Any ideas?

See next commit (and comment within).

**Comment by Hamms** [comment]
LGTM!

**Pull request merged**